ERROR Failed to process table market_actors: Invalid isoformat string: '201-10-01'

The bulk download is failing with the below error:
ValueError: Invalid isoformat string: '201-10-01'
ERROR Failed to process table market_actors: Invalid isoformat string: '201-10-01'

I got this error on 22nd Jan and also today.

Can you please help with this?

I get a similar error, maybe this gives more information. My download keeps going after this:

[ERROR] Error processing file 'Marktrollen.xml': 'Index(['MastrNummer'], dtype='object')'

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -17,6 +17,7 @@ and the versioning aims to respect [Semantic Versioning](http://semver.org/spec/
 ### Changed
 - Replace Marktrollen with MarktakteureUndRollen
   [#722](https://github.com/OpenEnergyPlatform/open-MaStR/pull/722)
+- Zero fill specific date strings [#728](https://github.com/OpenEnergyPlatform/open-MaStR/pull/728)
 
 ### Removed
 

--- a/open_mastr/xml_download/utils_write_to_database.py
+++ b/open_mastr/xml_download/utils_write_to_database.py
@@ -281,7 +281,10 @@ def cast_date_columns_to_string(xml_table_name: str, df: pd.DataFrame) -> pd.Dat
 
         if type(column[1].type) is Date:
             df[column_name] = (
-                df[column_name].dt.strftime("%Y-%m-%d").replace("NaT", None)
+                df[column_name]
+                .dt.strftime("%Y-%m-%d")
+                .replace("NaT", None)
+                .str.zfill(10)
             )
         elif type(column[1].type) is DateTime:
             df[column_name] = (

--- a/open_mastr/xml_download/utils_write_to_database.py
+++ b/open_mastr/xml_download/utils_write_to_database.py
@@ -280,12 +280,11 @@ def cast_date_columns_to_string(xml_table_name: str, df: pd.DataFrame) -> pd.Dat
         df[column_name] = pd.to_datetime(df[column_name], errors="coerce")
 
         if type(column[1].type) is Date:
-            df[column_name] = (
-                df[column_name]
-                .dt.strftime("%Y-%m-%d")
-                .replace("NaT", None)
-                .str.zfill(10)
-            )
+            mask = df[column_name].notna()
+            df[column_name] = df[column_name].dt.strftime("%Y-%m-%d")
+            df.loc[mask, column_name] = df.loc[mask, column_name].str.zfill(10)
+            df[column_name] = df[column_name].replace("NaT", None)
+
         elif type(column[1].type) is DateTime:
             df[column_name] = (
                 df[column_name].dt.strftime("%Y-%m-%d %H:%M:%S.%f").replace("NaT", None)

